### PR TITLE
Add loading spinner for causal policy tab

### PR DESCRIPTION
**Pull request by gaugup**
## Description

For large data the policy computation takes time. Hence, adding the loading spinner. 

image

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalAnalysisView.tsx [hunk 1]
```diff
@@ -171,4 +179,16 @@ export class CausalAnalysisView extends React.PureComponent<
...
+  private getPolicyElement(): any {
+    if (!this.state.isCausalPolicyDataLoading) {
+      return (
+        <TreatmentView
+          data={this.state.currentGlobalCausalPolicy}
+          telemetryHook={this.props.telemetryHook}
+        />
+      );
+    }
+    return <LoadingSpinner label={localization.Common.loading} />;
```
Comment (vinuthakaranth) [inline libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalAnalysisView.tsx:192]: nit: we can have conditional operator here
`return !this.state.isCausalPolicyDataLoading ? (
      <TreatmentView
        data={this.state.currentGlobalCausalPolicy}
        telemetryHook={this.props.telemetryHook}
      />
    ): <LoadingSpinner label={localization.Common.loading} />;`

File: libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalAnalysisView.tsx [hunk 2]
```diff
@@ -171,4 +179,16 @@ export class CausalAnalysisView extends React.PureComponent<
   ): ICausalAnalysisSingleData[] | undefined {
     return globalCausalEffects?.sort((d1, d2) => d2.point - d1.point);
   }
+
+  private getPolicyElement(): any {
```
Comment (vinuthakaranth) [inline libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalAnalysisView.tsx:183]: any can be replaced by React.ReactNode

**Comment by vinuthakaranth** [comment]
Design per se it would be good to have loading part just for the area below the dropdown instead of all tab reloading.
image

**Comment by vinuthakaranth** [comment]
> Design per se it would be good to have loading part just for the area below the dropdown instead of all tab reloading. image

discussed offline. It loads just once while loading tab and not while the selection of dropdown changes.

**Review by vinuthakaranth — APPROVED** [review]

File: libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalAnalysisView.tsx (hunk 1 above)
Comment (gaugup) [reply libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalAnalysisView.tsx:192]: The if statement has the same effect. Will leave the logic as is.

File: libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalAnalysisView.tsx (hunk 2 above)
Comment (gaugup) [reply libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalAnalysisView.tsx]: done

**Pull request merged**